src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -508,6 +516,62 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return resultExpression
         End Function
 
+        Private Shared Sub CheckRequiredMembersInObjectInitializer(
+            constructor As MethodSymbol,
+            initializers As ImmutableArray(Of BoundExpression),
+            creationSyntax As SyntaxNode,
+            diagnostics As BindingDiagnosticBag)
+
+            If constructor Is Nothing OrElse constructor.HasSetsRequiredMembers Then
+                Return
+            End If
+
+            If constructor.ContainingType.HasRequiredMembersError Then
+                ' A use-site diagnostic will be reported on the use, so we don't need to do any more checking here.
+                Return
+            End If
+
+            Dim requiredMembers = constructor.ContainingType.AllRequiredMembers
+
+            If requiredMembers.Count = 0 Then
+                Return
+            End If
+
+            Dim requiredMembersBuilder = requiredMembers.ToBuilder()
+
+            If Not initializers.IsDefaultOrEmpty Then
+                For Each initializer In initializers
+                    Dim assignmentOperator = TryCast(initializer, BoundAssignmentOperator)
+                    If assignmentOperator Is Nothing Then
+                        Continue For
+                    End If
+
+                    Dim memberSymbol As Symbol = If(
+                        DirectCast(TryCast(assignmentOperator.Left, BoundPropertyAccess)?.PropertySymbol, Symbol),
+                        TryCast(assignmentOperator.Left, BoundFieldAccess)?.FieldSymbol)
+
+                    If memberSymbol Is Nothing Then
+                        Continue For
+                    End If
+
+                    Dim requiredMember As Symbol = Nothing
+                    If Not requiredMembersBuilder.TryGetValue(memberSymbol.Name, requiredMember) Then
+                        Continue For
+                    End If
+
+                    If Not memberSymbol.Equals(requiredMember, TypeCompareKind.ConsiderEverything) Then
```

Comment:
>TypeCompareKind.ConsiderEverything [](http://example.com/codeflow?start=63&length=34)

Should probably use ```AllIgnoreOptionsForVB``` here, unless we can demonstrate significance of them with a test.